### PR TITLE
Chat: Support specifying priority for chatfilters

### PR DESCRIPTION
**Pull request by AnnikaCodes**
As per [this discussion](https://discord.com/channels/534298043176976395/581910956167266324/7841468372539146750).

### CODE REVIEW

**Comment by AnnikaCodes** [comment]
bump

**Comment by AnnikaCodes** [comment]
@Zarel bump (the test failures are unrelated)

**Comment by AnnikaCodes** [comment]
bump

File: server/chat-plugins/chat-monitor.ts [hunk 1]
```diff
@@ -320,55 +320,58 @@ void FS(MONITOR_FILE).readIfExists().then(data => {
 
 /* The sucrase transformation of optional chaining is too expensive to be used in a hot function like this. */
 /* eslint-disable @typescript-eslint/prefer-optional-chain */
-export const chatfilter: ChatFilter = function (message, user, room) {
```
Comment (Zarel) [inline server/chat-plugins/chat-monitor.ts:323]: One thing that would be interesting here (and also not change indentation level) would be:

```ts
export const chatfilter: ChatFilter = function (message, user, room) {
  ...
};
chatfilter.priority = 0;
```

Which can be left off for most chatfilters at default priority.

Comment (Zarel) [reply server/chat-plugins/chat-monitor.ts:323]: I would leave it up to you, but I would definitely strongly prefer that approach.

Comment (AnnikaCodes) [reply server/chat-plugins/chat-monitor.ts]: That's definitely simpler.

Comment (AnnikaCodes) [reply server/chat-plugins/chat-monitor.ts]: I'll change to that

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**